homeassistant/components/switchbot/__init__.py

```diff
@@ -83,6 +86,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     hass.data[DOMAIN][entry.entry_id] = {DATA_COORDINATOR: coordinator}
 
+    if entry.data[CONF_SENSOR_TYPE] == ATTR_BOT:
+        PLATFORMS.remove("cover")
+
+    if entry.data[CONF_SENSOR_TYPE] == ATTR_CURTAIN:
+        PLATFORMS.remove("switch")
+
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Comment:
```suggestion
    hass.config_entries.async_setup_platforms(entry, PLATFORMS_BY_TYPE[sensor_type])
```

Reply:
You'll need something similar in `async_unload_entry `

Reply:
Done. Also added in `async_unload_entry` .